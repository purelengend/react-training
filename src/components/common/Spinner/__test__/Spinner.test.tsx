import { renderer } from '@src/test/test-utils';

import { Spinner } from '..';

describe('Spinner test case', () => {
  it('should render correctly without custom class name', () => {
    const mockSpinner = renderer(<Spinner />);

    expect(mockSpinner.toJSON()).toMatchSnapshot();
  });

  it('should accept custom class name', () => {
    const mockSpinner = renderer(<Spinner customStyle="mock-class-name" />);

    expect(mockSpinner.toJSON()).toMatchSnapshot();
  });
});
