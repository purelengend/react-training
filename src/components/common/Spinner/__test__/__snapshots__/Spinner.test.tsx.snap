// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Spinner test case > should accept custom class name 1`] = `
<div
  className="_loader_3c694b mock-class-name"
  id="spin"
/>
`;

exports[`Spinner test case > should render correctly without custom class name 1`] = `
<div
  className="_loader_3c694b undefined"
  id="spin"
/>
`;
